algorithm-registry.md

```diff
@@ -0,0 +1,64 @@
+# Algorithm Registry
+
+This file is designed to act as a source of truth regarding what signing
+algorithms are recommended across the Sigstore ecosystem. Any changes to this
+file **must** be reflected in the `PublicKeyDetails` enumeration in
+[`sigstore_common.proto`] in [sigstore/protobuf-specs].
+
+Sigstore clients aren't required to support all algorithms in this registry,
+and **MAY** support algorithms that aren't in the registry. However,
+compatibility with the Sigstore Public Good Instance requires support
+for at least one of these algorithms.
```

Comment:
NB: I strengthened the language here; the current doc in `protobuf-specs` says:

```
Note that Sigstore clients and services aren't required support all algorithms in this registry and may support algorithms that aren't in the registry. The algorithm registry is more of a guideline than a rule and is meant to serve as a secure set of defaults that the community can follow.
```

Reply:
should we add that cross language client support may require supporting all these algorithms?

Reply:
Works for me, although I _think_ that kind of language could also go in the client spec (which I'm going to start updating once this is merged). WDYT?

Reply:
Works for me